src/P2P/Node.hs

```diff
@@ -350,7 +396,10 @@ syncFromPeer node info = runClientM sync env >>= \case
             logg node Warn $ "failed to sync peers from " <> showInfo info <> ": " <> sshow e
             return False
     Right p -> do
-        goods <- wither (removeTrivialPeers node) $ _pageItems p
+        goods <- fmap catMaybes
+            $ mapConcurrently (guardPeerDbOfNode node)
+            $ filter (\i -> me /= _peerId i)
+            $ _pageItems p
```

Comment:
We expect this page to be small, yeah? So the `mapConcurrently` won't trash the runtime?